crates/rolldown_binding/src/types/binding_outputs.rs

```diff
@@ -32,45 +38,55 @@ impl BindingOutputs {
   pub fn assets(&mut self) -> Vec<BindingOutputAsset> {
     let mut assets: Vec<BindingOutputAsset> = vec![];
 
-    self.inner.iter_mut().for_each(|o| match o {
-      rolldown_common::Output::Asset(asset) => {
-        assets.push(BindingOutputAsset::new(unsafe { std::mem::transmute(asset.as_mut()) }));
-      }
-      rolldown_common::Output::Chunk(_) => {}
+    self.inner.with_inner(|inner| {
+      let mut inner = inner.lock().unwrap();
+      inner.iter_mut().for_each(|o| match o {
+        rolldown_common::Output::Asset(asset) => {
+          assets.push(BindingOutputAsset::new(unsafe { std::mem::transmute(asset.as_mut()) }));
```

Comment:
We could remove this `unsafe` by following how this PR did to `BindingOutputChunk`. But I still wondering if there are better to model this.